text/batch-l2.md

```diff
@@ -0,0 +1,1540 @@
+`'B'` jobs will no longer be considered after 5 minutes.
+
+The following code specifies that the weight factors will no longer be considered after 5 minutes:
+
+```ts
+const fairsharePolicy = new FairshareSchedulingPolicy(this, 'myFairsharePolicy', {
+   shareDecay: Duration.seconds(600),
+});
+```
+
+If you have high priority jobs that should always be executed as soon as they arrive,
+you can define a `computeReservation` to specify what percentage of the
+maximum vCPU capacity should be reserved for those jobs. For example, if you specify a `computeReservation` of 75, then
+75% of the maxvCPUs of the Compute Environment are reserved for all the share identifiers in this scheduling policy.
+The following example specifies that 75% of the compute capacity should be reserved for `'A'`, `'B'`, or `'C'` jobs:
+
+```ts
+const fairsharePolicy = new FairshareSchedulingPolicy(this, 'myFairsharePolicy', {
+  computeReservation: 75,
+  shares: [{
+    shareIdentifier: 'A',
+  }],
+});
+
+fairsharePolicy.addShare({
+  shareIdentifier: 'B',
+});
+
+fairsharePolicy.addShare({
+  shareIdentifier: 'C',
+});
+```
+
+The `computeReservation` is **not** split between shares.
+That is, it guarantees that 75% of the maximum vCPUs are reserved for jobs with share identifier `'A'` or `'B'` `'C'`;
+it does **not** guarantee that 25% is allocated to `'A'` and 25% is allocated to `'B'` and 25% is allocated to `'C'`.
+Instead, it is possible for `'A'` jobs to fill the entire queue and starve the `'B'` and `'C'` jobs. To avoid this,
+define multiple `FairshareSchedulingPolicy`s:
+
+```ts
+new FairshareSchedulingPolicy(this, 'AFairsharePolicy', {
+  computeReservation: 25,
+  shares: [{
+    shareIdentifier: 'A',
+  }],
+});
+new FairshareSchedulingPolicy(this, 'BFairsharePolicy', {
+  computeReservation: 25,
+  shares: [{
+    shareIdentifier: 'B',
+  }],
+});
+new FairshareSchedulingPolicy(this, 'CFairsharePolicy', {
+  computeReservation: 25,
+  shares: [{
+    shareIdentifier: 'C',
+  }],
+});
+```
+
+The above example ensures that each share can take no more than 50% of the total capacity
+(25% reserved + 25% unreserved).
+
+### Configuring Job Retry Policies
+
+Certain workflows may result in Jobs failing due to intermittent issues.
+Jobs can specify retry policies to respond to different failures with different actions.
+There are three different ways information about the way a Job exited can be conveyed;
+
+* `exitCode`: the exit code returned from the process executed by the container. Will only match non-zero exit codes.
+* `reason`: any middleware errors, like your Docker registry being down.
+* `statusReason`: infrastructure errors, most commonly your spot instance being reclaimed. 
+
+You can specify a glob string to match each of these and react to different failures accordingly.
+Up to five different retry strategies can be configured for each Job,
+and each strategy can match against some or all of `exitCode`, `reason`, and `statusReason`.
```

Comment:
From the description above, I'm not sure whether it makes sense to match on multiple of these together. Does it? Can BOTH my instance be reclaimed AS WELL AS my process exits with an exitCode?

I'd rather see an enum-like class that is more reasonable and hides the distinction.

```
retryStrategies [
{
  action: Action.RETRY,
  reason: Reason.SPOT_INSTANCE_RECLAIMED,
},
{
  action: Action.EXIT,
  reason: Reason.CANNOT_PULL_CONTAINER,
},
{
  action: Action.RETRY,
  reason: Reason.nonZeroExitCode(),
},
{
  action: Action.Exit,
  reason: Reason.custom({
    onExitCode: '40*', // This is so weird
    onStatusReason: 'bananas',
  }),
},
]
```

Reply:
And I also don't particularly care for the list of `{ action, reason }` pairs, but that's necessary if order is important.

Reply:
Order is important here, batch will perform the first `action` whose `reason` matches

Reply:
I'm not sure why this is possible, but it's most common to use just one at a time, so this is the better UX